extensions/job_management/standard/recommendations/job-management/create/PER_content-schema.adoc

```diff
@@ -0,0 +1,7 @@
+[[per_job-management_create_content-schema]]
+[permission]
+====
+[%metadata]
+label:: /per/job-management/create-content-schema
+part:: The `Content-Schema` header MAY be an URI to a JSON or OpenAPI 3.0 Schema document that describes the structure of the request body.
+====
```

Comment:
Can someone explain to me, how this is meant to work? I'm lacking some context...
How is this different to the OpenAPI that is linked to from the landing page? Which HTTP methods and endpoint is this header delivered with? HEAD /jobs? OPTIONS /jobs? ...

Reply:
The idea is that a client would send a request similar to the following: 
```http
POST /jobs HTTP/1.1
Content-Type: application/json
Content-Schema: https://raw.githubusercontent.com/common-workflow-language/cwl-v1.2/main/json-schema/cwl.yaml

{"cwl-workflow": "..."}
```
```http
POST /jobs HTTP/1.1
Content-Type: application/json
Content-Schema: https://api.openeo.org/assets/pg-schema.json

{"openeo-process-graph": "..."}
```
```http
POST /jobs HTTP/1.1
Content-Type: application/json
Content-Schema: https://raw.githubusercontent.com/opengeospatial/ogcapi-processes/refs/heads/master/openapi/schemas/processes-workflows/execute-workflows.yaml

{"oap-ad-hoc-workflow": "..."}
```

This would allow the server to (explicitly) distinguish between different contents submitted with similar JSON (or YAML, etc.) types, while giving the flexibility to the server to decide which content schema it prefers to use by default, and does not introduce modifications to `Content-Type` that could break existing implementations assuming only `application/json`.

A server that is already more *OGC API - Processes - Workflows*, CWL or openEO aligned can then continue to submit its contents as before, and has a way to indicate the alternate `Content-Schema` if it wants to support another. `Content-Schema` could be provided all the time, but would be ignored by servers that only have one representation supported.

In every case, a mismatch of `Content-Schema`/parsing would raise HTTP 422 or similar error codes indicating it could not understand it, but having `Content-Schema` explicitly helps the server provide a clearer message and informed decision about "why" the contents are not processable. 

`Content-Schema` could be used on other endpoints as well, such as when requesting `GET /jobs/{jobId}/run` or `GET /processes/{processId}/package` to describe the various representations that could be returned by those responses.

Reply:
I think there should be more alignment then just the endpoint name and then specifying which type it will be. 
I also think just submitting the workflow without additional information is not very extensible.

```
POST /jobs HTTP/1.1
Content-Type: application/json

{"title": "ABC", "processType": "openeo", "process": { openEO process }}
```

or CWL:

```
POST /jobs HTTP/1.1
Content-Type: application/yaml

title: ABC
processType: cwl
process:
  cwl process
```

etc.